ydb/core/base/auth.cpp

```diff
@@ -0,0 +1,43 @@
+#include "auth.h"
+
+namespace NKikimr {
+
+namespace {
+
+bool HasToken(const TAppData* appData, const NACLib::TUserToken& userToken) {
+    for (const auto& sid : appData->AdministrationAllowedSIDs) {
+        if (userToken.IsExist(sid)) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+}
+
+bool IsAdministrator(const TAppData* appData, const TString& userToken) {
+    if (appData->AdministrationAllowedSIDs.empty()) {
+        return true;
+    }
+
+    if (!userToken) {
+        return false;
+    }
+
+    return HasToken(appData, NACLib::TUserToken(userToken));
```

Comment:
Minor note: `NACLib::TUserToken()` will cause process crash if `userToken` contains invalid (or empty) protobuf. It was ok when token string were getting retrieved from the known context where token string guaranteed to be valid. But utility method lacks that guarantee.